src/beautifiers/rubocop.coffee

```diff
@@ -17,52 +18,54 @@ module.exports = class Rubocop extends Beautifier
   }
 
   beautify: (text, language, options) ->
+    editor = atom?.workspace?.getActiveTextEditor()
+    if editor?
+      fullPath = editor.getPath()
+      [projectPath, relativePath] = atom.project.relativizePath(fullPath)
+    else
+      throw new Error("No active editor found!")
+
+    # Find the rubocop path
+    rubocopPath = "rubocop"
     @Promise.all([
       @which(options.rubocop_path) if options.rubocop_path
       @which('rubocop')
-    ]).then((paths) =>
+    ])
+    .then((paths) =>
       @debug('rubocop paths', paths)
-      _ = require 'lodash'
-      path = require 'path'
       # Get first valid, absolute path
-      rubocopPath = _.find(paths, (p) -> p and path.isAbsolute(p) )
+      rubocopPath = paths.find((p) -> p and path.isAbsolute(p))
       @verbose('rubocopPath', rubocopPath)
       @debug('rubocopPath', rubocopPath, paths)
+    )
 
-      configFile = path.join(atom.project.getPaths()[0], ".rubocop.yml")
-
-      fs = require 'fs'
+    # Find or generate a config file if non exists
+    configFile = @findFile(path.dirname(fullPath), ".rubocop.yml")
+    if !configFile?
+      yaml = require("yaml-front-matter")
+      config = {
+        "Style/IndentationWidth":
+          "Width": options.indent_size
+      }
+      tempConfig = @tempFile("rubocop-config", yaml.safeDump(config))
```

Comment:
Should `tempConfig` be `configFile`? I do not see where `configFile` is used.

Reply:
No, `configFile` is just use for the if statement so the code is cleaner. I think `if !configFile?` is more self-documenting than `if !@findFile(path.dirname(fullPath), ".rubocop.yml")?`, but I will change it by your request.

Reply:
Fixed.

Reply:
Oh no I agree with `configFile` being more readable code! Sorry for not being clear, please change it back 😄 .

What I was trying to find out is how it works in the case `configFile` is truthy -- when `.rubocop.yml` does exist. How does Rubocop get that `.rubocop.yml` file if `configFile` is not used later in the code?

Reply:
I've changed `configFile` back!

I hope you're sitting down for the answer to the `.rubocopy.yml` question!

Rubocop will traverse the tree starting at the file path in question, looking for `.rubocop.yml` files, stopping at the project root (I think). All of those `.rubocop.yml` files will be loaded, with the ones closest to the file in the tree taking precedence. In addition to that, Rubocop configuration files can each inherit from multiple other local, gem, or even remote `.rubocop.yml`. In addition to this, configuration files can `Exclude` or `Include` certain path patterns, which are dependent on the location where the configuration file is. 

The end result is a convoluted config system that is very difficult to reason about (even for Rubocop itself). This setup expects files to be in the exact same position with the same name, or a bunch of filters and settings break. 

If you attempt to override all this with the `--config` switch, it ignores all of this and uses only that one config file, and on top of that there's a bug in that it uses relative paths from the configuration path instead of the working dir, which I linked above. This is why there were a bunch of subtle issues with the previous implementation.

Reply:
😆 Thanks for the warning and the proceeding explanation!

In summary, the code will do the following:
1. Look for possible config files `Rubocop` might use, call the first one found `configFile`.
2. If there is not a `configFile` found, then create a new one with Atom-Beautify's settings, and call this `tempConfig`.
3. Pass the `relativePath` to Rubocop. Note, it may be empty (i.e. `undefined`) and therefore will be removed from argument list.
4. If `tempConfig` is set -- use Atom-Beautify's custom Rubocop config file -- then pass the argument to Rubocop
5. Run Rubocop with arguments, removing the arguments which are empty (i.e. `undefined`), such as possibly `relativePath`. If `relativePath` is empty, then Rubocop will not have a path to attempt to find `.rubocop.yml` files. Otherwise, if `relativePath` is set -- file exists -- then Rubocop can attempt to find `.rubocop.yml` files for configuration.

Gotcha! 👍 Thanks!

Reply:
Sort of! 

1. ~~Look for possible config files Rubocop might use, call the first one found configFile.~~ Look in the project path for a `.rubocop.yml`. This doesn't tree-traverse from the file location the way Rubocop does. We *could* do that, which would prevent a (probably very rare) bug where someone has a `.rubocop.yml` file deeper in the tree structure, but *not* in the project root where it is commonly placed...
2. If there is not a configFile found, then create a new one with Atom-Beautify's settings, and call this tempConfig. 👍
3. ~~Pass the relativePath to Rubocop. Note, it may be empty (i.e. undefined) and therefore will be removed from argument list.~~ `relativePath` was just already there since it was pulled from Atom, so I passed it in to `--stdin` (it made debugging output easier to read). We don't actually care what that parameter is, so I changed it to use a hard coded value which solves the `undefined` `context.filePath` problem.
1. If tempConfig is set -- use Atom-Beautify's custom Rubocop config file -- then pass the argument to Rubocop 👍
1. ~~Run Rubocop with arguments, removing the arguments which are empty (i.e. undefined), such as possibly relativePath. If relativePath is empty, then Rubocop will not have a path to attempt to find .rubocop.yml files. Otherwise, if relativePath is set -- file exists -- then Rubocop can attempt to find .rubocop.yml files for configuration.~~ Rubocop will look up config files based on where the `rubocop` command is run from.